packages/near-membrane-base/src/blue.ts

```diff
@@ -161,52 +152,92 @@ export function blueProxyFactory(env: MembraneBroker) {
         // istanbul ignore else
         if ('writable' in blueDescriptor) {
             // We are dealing with a value descriptor.
-            const { value } = blueDescriptor;
-            blueDescriptor.value =
-                typeof value === 'function'
-                    ? // We are dealing with a method (optimization).
-                      getBlueFunction(value)
-                    : getBlueValue(value);
+            blueDescriptor.value = getBlueValue(blueDescriptor.value);
         } else {
             // See additional details above.
             // We are dealing with accessors.
             const { get, set } = blueDescriptor;
-            if (typeof get === 'function') {
-                blueDescriptor.get = getBlueFunction(get);
+            if (get !== undefined) {
+                blueDescriptor.get = getBlueValue(get);
             }
-            if (typeof set === 'function') {
-                blueDescriptor.set = getBlueFunction(set);
+            if (set !== undefined) {
+                blueDescriptor.set = getBlueValue(set);
             }
         }
         return blueDescriptor;
     }
 
     function getBlueFunction(redFn: RedFunction): BlueFunction {
-        const blueFn = env.getBlueRef(redFn);
-        if (blueFn === undefined) {
-            return createBlueProxy(redFn) as BlueFunction;
+        // caching logic
+        const blueFn = WeakMapGet(env.redMap, redFn) as RedFunction | undefined;
+        if (blueFn !== undefined) {
+            return blueFn;
+        }
+        // extracting the metadata about the proxy target
+        let targetTraits = TargetTraits.IsFunction;
+        let targetFunctionName: string | undefined;
+        // detecting arrow function vs function
+        try {
+            targetTraits |= (!('prototype' in redFn) ? 1 : 0) << 3; // IsArrowFunction
```

Comment:
this is another example of typescript failing to provide a real way to use bitwise (or maybe I just don't know how to use TS jajajaja). Basically, the position of the 0 or 1 bit can't be inferred from the type itself, forcing me to have fixed position that must be in sync with the type declaration, otherwise chaos will reign.

Reply:
this is happening in 3 different places per color.

Reply:
(I'm not sure what chaos you were experiencing)

While reviewing this locally, I can write the assignment as: 

```js
targetTraits |= +!('prototype' in redFn) && TargetTraits.IsArrowFunction;
```

Reply:
jaja! this is awesome... big oversight from my side, those types are actionable, so this clearly solve the chaos that I was referencing too, it turns out that typescript does a fine job for bitwise via enums, thanks for the suggestion.